src/dynamics/ads_player/ads_player.js

```diff
@@ -381,34 +377,52 @@ Scoped.define("module:Ads.Dynamics.Player", [
                     })
                 },
 
-                setEndCardBackground: function(width, height) {
+                captureAdEndCardBackground: function(width, height) {
                     const ad = this.get("ad");
                     if (ad) {
-                        this.updateEndCardImage(ad, width, height);
+                        this.generateEndCardImage(ad, width, height);
                     }
                 },
 
-                updateEndCardImage: function(ad, width, height) {
+                setEndCardBackground: function(width, height) {
+                    if (width && height) {
+                        if (this.shouldShowFirstFrameAsEndcard()) {
+                            this.captureAdEndCardBackground(width, height);
+                            if (this._src) {
+                                this.getAdContainer().style.backgroundImage = `url("${this._src}")`;
+                            }
+                        }
+                    }
+                },
+
+                generateEndCardImage: function(ad, width, height) {
                     if (this._video && this._canvas && this._mediaUrl) {
                         this.resizeCanvas(width, height);
                         return;
                     }
                     this._video = document.createElement("video");
                     this._canvas = document.createElement("canvas");
                     this._mediaUrl = ad?.data?.mediaUrl;
-                    this._video.crossOrigin = "anonymous";
-                    this._video.src = this._mediaUrl;
-                    this._video.muted = true;
-                    this._video.play();
-                    setTimeout(function() {
-                        this._canvas.width = width;
-                        this._canvas.height = height;
-                        this._canvas
-                            .getContext("2d")
-                            .drawImage(this._video, 0, 0, this._canvas.width, this._canvas.height);
-                        this._src = this._canvas.toDataURL("image/png");
-                        this._video.pause();
-                    }.bind(this), 1000);
+                    if (this._mediaUrl) {
+                        fetch(this._mediaUrl)
+                            .then(response => response.blob())
+                            .then(blob => {
+                                this._video.src = URL.createObjectURL(blob);
+                                this._video.crossOrigin = "anonymous";
+                                this._video.muted = true;
+                                this._video.play();
+                                setTimeout(function() {
```

Comment:
Hey @EdmundsBankai looks great, only what you think to replave setTimeout via Promise. As timeout is looks like something hard coded way. But with promise you sure it's played draw and paused after that. Also catch if there's any play error occurs.  
```
this._video.play().then(() => {
 // logic here and pause at the end
 this._video.pause();
}).catch()
```

Reply:
@sambua so, the idea is to capture a screenshot after 1 second into video. so would you recommend something like
```javascript
...
new Promise(resolve => {
            setTimeout(() => {
                this._canvas.width = width;
                this._canvas.height = height;
                this._canvas
                    .getContext("2d")
                    .drawImage(this._video, 0, 0, this._canvas.width, this._canvas.height);
                this._src = this._canvas.toDataURL("image/png");
                this._video.pause();
                resolve(); // Resolve the Promise after the timeout
            }, 1000);
            ...
```

Reply:
No @EdmundsBankai `this._video.play()` itself is Promise. So no need for the timeout. Full code could like 
```
    fetch(this._mediaUrl)
        .then(response => response.blob())
        .then(blob => {
            this._video.src = URL.createObjectURL(blob);
            this._video.crossOrigin = "anonymous";
            this._video.muted = true;
            return this._video.play(); // << will return Promise
        })
        .then(() => {
            this._canvas.width = width;
            this._canvas.height = height;
            this._canvas
                .getContext("2d")
                .drawImage(this._video, 0, 0, this._canvas.width, this._canvas.height);
            this._src = this._canvas.toDataURL("image/png");
            this._video.pause(); 
        }).catch((e) => console.log(`Error: ${e}`));

```

Reply:
@sambua updated